### PR TITLE
deps: normalize dependencies

### PR DESCRIPTION
**Pull request by diegomarquezp**

### CODE REVIEW

**Review by blakeli0 — APPROVED** [review]

File: api-common-java/pom.xml [hunk 1]
```diff
@@ -80,6 +80,11 @@
             <version>1.4.2</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+          <groupId>com.google.j2objc</groupId>
+          <artifactId>j2objc-annotations</artifactId>
+          <version>${j2objc-annotations.version}</version>
```
Comment (blakeli0) [inline api-common-java/pom.xml:86]: @suztomo We have to declare this dependency here  and the version in `gapic-generator-java-pom-parent`, to resolve a Maven enforcer failure. In addition, it is already declared in third-party-dependencies, do you have a good way to declare this dependency in a centralized place?

Comment (suztomo) [reply api-common-java/pom.xml:86]: Use variables. Follow gRPC's version.

File: java-shared-dependencies/third-party-dependencies/pom.xml [hunk 2]
```diff
@@ -37,7 +37,7 @@
     <!-- ensure checker-qual version matches what Guava uses -->
     <checker-qual.version>3.42.0</checker-qual.version>
     <perfmark-api.version>0.27.0</perfmark-api.version>
-    <j2objc-annotations.version>2.8</j2objc-annotations.version>
```
Comment (suztomo) [inline java-shared-dependencies/third-party-dependencies/pom.xml:40]: Shall we keep this as old version for now, to reduce uncertainty?

File: gapic-generator-java-pom-parent/pom.xml [hunk 3]
```diff
@@ -30,10 +30,12 @@
     <google.auth.version>1.23.0</google.auth.version>
     <google.http-client.version>1.44.1</google.http-client.version>
     <gson.version>2.10.1</gson.version>
-    <guava.version>32.1.3-jre</guava.version>
```
Comment (suztomo) [inline gapic-generator-java-pom-parent/pom.xml:33]: Can you check the release note for Guava 33 for any breaking changes?

Comment (diegomarquezp) [reply gapic-generator-java-pom-parent/pom.xml:33]: Sure, I just did. I did not [find](https://github.com/google/guava/releases) any breaking changes

File: java-shared-dependencies/third-party-dependencies/pom.xml (hunk 2 above)
Comment (diegomarquezp) [reply java-shared-dependencies/third-party-dependencies/pom.xml:40]: It was necessary as part of updating `guava` to `33.1.0`. Setting it to 2.8 causes:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.4.1:enforce (enforce) on project grpc-google-common-protos: 
[ERROR] Rule 2: org.apache.maven.enforcer.rules.dependency.RequireUpperBoundDeps failed with message:
[ERROR] Failed while enforcing RequireUpperBoundDeps. The error(s) are [
[ERROR] Require upper bound dependencies error for com.google.j2objc:j2objc-annotations:2.8 paths to dependency are:
[ERROR] +-com.google.api.grpc:grpc-google-common-protos:2.36.1-SNAPSHOT
[ERROR]   +-com.google.guava:guava:33.1.0-jre
[ERROR]     +-com.google.j2objc:j2objc-annotations:2.8 (managed) <-- com.google.j2objc:j2objc-annotations:3.0.0
[ERROR] ]
[ERROR] -> [Help 1]
```

File: api-common-java/pom.xml (hunk 1 above)
Comment (diegomarquezp) [reply api-common-java/pom.xml:86]: We are using variables for `errorprone` and `j2objc` in
 - gapic-generator-java-pom-parent
 - java-shared-dependencies/third-party-dependencies

File: gapic-generator-java-pom-parent/pom.xml (hunk 3 above)
Comment (suztomo) [reply gapic-generator-java-pom-parent/pom.xml:33]: Thank you

File: java-shared-dependencies/third-party-dependencies/pom.xml (hunk 2 above)
Comment (suztomo) [reply java-shared-dependencies/third-party-dependencies/pom.xml:40]: OK. Thank you for checking.

**Review by suztomo — APPROVED** [review]

**Pull request merged**